lib/DxilValidation/DxilContainerValidation.cpp

```diff
@@ -73,6 +78,522 @@ struct DiagRestore {
 static void emitDxilDiag(LLVMContext &Ctx, const char *str) {
   hlsl::dxilutil::EmitErrorOnContext(Ctx, str);
 }
+
+// Helper class which build PSV ViewID state from serialized view id state.
+struct SimpleViewIDState {
+  unsigned NumInputSigScalars = 0;
+  unsigned NumOutputSigScalars[DXIL::kNumOutputStreams] = {0, 0, 0, 0};
+  unsigned NumPCOrPrimSigScalars = 0;
```

Comment:
These are all three stored as scalars, but every one of them is aligned by 4 and converted to vectors every time they are retrieved. The only exception is the constructor where they are originally assigned where that original value is readily available. I think it would be cleaner to store these as vectors rather than perform that conversion every time.